src/test/util/event.rs

```diff
@@ -183,6 +183,16 @@ impl EventClient {
         }
         panic!("could not find event for {} command", command_name);
     }
+
+    pub fn topology(&self) -> String {
```

Comment:
Sorry, must've missed responding to this one; using a `String` here is a bit simpler because the topology types in the tests ("sharded", "replicaset", "single") don't correspond directly to the `TopologyType` types (`Single`, `ReplicaSetNoPrimary`, `ReplicaSetWithPrimary`, `Sharded`, `Unknown`), so there isn't a direct translation from "replicaset" in the test file into a `TopologyType`. I suppose I could just choose one of the replica set types to use for serializing into a `RunOn`/returning from this method? Or maybe introduce a separate test-specific `TopologyType` enum for the three options that the json provides.

Reply:
Hm, I see. Probably just sticking with strings is fine then.